src/pytest_workflow/content_tests.py

```diff
@@ -33,72 +33,42 @@
 
 
 def check_content(strings: Iterable[str],
-                  text_lines: Iterable[str]) -> Set[str]:
+                  patterns: Iterable[str],
+                  text_lines: Iterable[str]):
     """
-    Checks whether any of the strings is present in the text lines
+    Checks whether any of the strings or patterns is present in the text lines
     It only reads the lines once and it stops reading when
-    everything is found. This makes searching for strings in large bodies of
-    text more efficient.
-    :param strings: A list of strings for which the present is checked
+    everything is found. This makes searching for strings  and patterns in
+    large bodies of text more efficient.
+    :param strings: A list of strings to check for
+    :param patterns: A list of regex patterns to check for
     :param text_lines: The lines of text that need to be searched.
-    :return: A tuple with a set of found strings, and a set of not found
-    strings
+    :return: A tuple with a set of found strings, and a set of found patterns.
     """
-
-    # Create two sets. By default all strings are not found.
     strings_to_check = set(strings)
     found_strings: Set[str] = set()
+    regex_to_match: Set[re.Pattern] = {re.compile(pattern)
+                                       for pattern in patterns}
+    found_regexes: Set[re.Pattern] = set()
 
     for line in text_lines:
         # Break the loop if all strings are found
         # Python implements fast set equality checking by checking length first
-        if found_strings == strings_to_check:
+        if not strings_to_check and not regex_to_match:
```

Comment:
The nature of this check changed and the comment is now not reflecting the purpose of the if statement any more. I believe something went wrong here because the previous check made an early exit from the loop once everything that was expected was found. The new statement broke the previous early exit. The check that is done here should probably be done before the for loop even starts. Or even before the files are opened.

Reply:
> The nature of this check changed and the comment is now not reflecting the purpose of the if statement any more.

Good catch. I will remove the fast equality checking comment.

> I believe something went wrong here because the previous check made an early exit from the loop once everything that was expected was found. The new statement broke the previous early exit.

If you read along you will see the following code:
```
        strings_to_check -= found_strings
        regex_to_match -= found_regexes
```
The strings_to_check and regex_to_match are emptied along the search. If everything is found these sets will be completely empty. The statement will then evaluate to true and the loop will break. The rest of the file will not be read. 

> The check that is done here should probably be done before the for loop even starts. Or even before the files are opened.

If there are no content tests (must_contain, must_not_contain, must_contain_regex, must_not_contain_regex) the ContentTestCollector is not called. So there is no case where a file is opened when there are no strings to check. 
See https://github.com/LUMC/pytest-workflow/blob/75de0664922ed7c9992deb7cde43e82f5991b037/src/pytest_workflow/file_tests.py#L63 .

Reply:
> The strings_to_check and regex_to_match are emptied along the search. If everything is found these sets will be completely empty. The statement will then evaluate to true and the loop will break. The rest of the file will not be read.

Ooops, didn't read it carefully enough. I also see now why it is this way - the search space is reduced with every found match.

Reply:
Indeed. The performance is quite okay because of that too. I just did a check on a large file to check if the search indeed exits early and it all seems to work fine.